src/components/CDNUploadModal.tsx

```diff
@@ -15,98 +27,59 @@ export const defaultCdn = {
     }) as () => void,
 };
 
-interface GitHubOption {
-    type: "dir" | "file",
-    label: string;
-    value: string;
-}
-const AlwaysVisibleInput = (props: any) => <components.Input {...props} isHidden={false} />;
-
-interface GitHubDirInputProps {
-    repo: GitHubRepository;
-    className: string;
-    dir: string | undefined;
-    setDir: (newDir: string | undefined) => void;
-    invalid?: boolean;
-}
-const GitHubDirInput = ({repo, className, dir, setDir, invalid}: GitHubDirInputProps) => {
-
-    // Hacking react-select to do sensible things reading list:
-    //  - https://github.com/JedWatson/react-select/issues/1558#issuecomment-738880505
-    //  - https://github.com/JedWatson/react-select/discussions/4302
-    //  - https://stackoverflow.com/questions/51951379/how-do-i-implement-field-validation-for-react-select
-
-    const appContext = useContext(AppContext);
-    const [pathOption, setPathOption] = useState<GitHubOption | null>();
-    const searchablePathPart = dir ? dir.split("/").length === 0 ? dir : dir.split("/").slice(0, -1).join("/") : "";
-
-    const {data, isValidating, error} = useSWR(contentsPath(searchablePathPart ?? "", appContext.github.branch, repo));
-
-    const pathOptions: GroupBase<GitHubOption>[] = useMemo(() => {
-        return (error || !data) ? [] : [{
-            options: data?.type === "file" ? {value: data.path, label: data.path, isDisabled: true} : data?.map((d: any) => ({value: d.path, label: d.path, isDisabled: d.type === "file"})) ?? [],
-            label: "Existing directories"
-        }];
-    }, [error, data]);
-
-    const formatCreateLabel = (path: string) => `Create directory "${path.replace(/\/$/, "")}"`;
-
-    const onInputChange = (inputValue: string, { action }: InputActionMeta) => {
-        // onBlur => setInputValue to last selected value
-        if (action === "input-blur") {
-            setDir(pathOption ? pathOption.value : "");
+const renameCDNFile = async (context: ContextType<typeof AppContext>, item: PopupEntry, setSelection?: (selection: Selection) => void) => {
+    let newName = window.prompt("Please type a new name for the file. If no extension is provided, \".json\" will be assumed", item.name);
+    if (newName) {
+        // Don't both renaming a file to it's previous name
+        if (item.path.replace(dirname(item.path) + "/", "") === newName) {
+            return;
         }
-        // onInputChange => update inputValue
-        else if (action === "input-change") {
-            setDir(inputValue);
+        // Ensure the file type (extension) stays the same
+        if (ext(newName) !== ext(item.name)) {
+            alert("Please don't modify the file extension! Cancelling rename operation...");
```

Comment:
This is because we have a whitelist of filetypes that content can upload to the CDN - if this were allowed it would be quite easy to circumvent that whitelist just using the UI. 
They could just go via GitHub and add whatever they wanted anyway, but I wanted the upload modal at least to be watertight 🤷